clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -8141,16 +8141,22 @@ void TCETargetCodeGenInfo::setTargetAttributes(
 
         SmallVector<llvm::Metadata *, 5> Operands;
         Operands.push_back(llvm::ConstantAsMetadata::get(F));
-
-        Operands.push_back(
-            llvm::ConstantAsMetadata::get(llvm::Constant::getIntegerValue(
-                M.Int32Ty, llvm::APInt(32, Attr->getXDim()))));
-        Operands.push_back(
-            llvm::ConstantAsMetadata::get(llvm::Constant::getIntegerValue(
-                M.Int32Ty, llvm::APInt(32, Attr->getYDim()))));
-        Operands.push_back(
-            llvm::ConstantAsMetadata::get(llvm::Constant::getIntegerValue(
-                M.Int32Ty, llvm::APInt(32, Attr->getZDim()))));
+        unsigned XDim = Attr->getXDim()
+                            ->EvaluateKnownConstInt(M.getContext())
+                            .getExtValue();
+        unsigned YDim = Attr->getYDim()
+                            ->EvaluateKnownConstInt(M.getContext())
+                            .getExtValue();
+        unsigned ZDim = Attr->getZDim()
+                            ->EvaluateKnownConstInt(M.getContext())
+                            .getExtValue();
+
+        Operands.push_back(llvm::ConstantAsMetadata::get(
+            llvm::Constant::getIntegerValue(M.Int32Ty, llvm::APInt(32, XDim))));
```

Comment:
Similar question here about whether this should be creating `UInt32Ty` values.

Reply:
Done